src/proc/layouter.rs

```diff
@@ -137,40 +202,27 @@ impl Layouter {
             let layout = match ty.inner {
                 Ti::Scalar { width, .. } | Ti::Atomic { width, .. } => TypeLayout {
                     size,
-                    alignment: Alignment::new(width as u32)
-                        .ok_or(LayoutErrorInner::ZeroWidth.with(ty_handle))?,
```

Comment:
I was planning to change the `width` in the IR from a `u8` to either a `PowerOfTwoU8` (concrete type or create some generic POT type used by `width`s and the `Alignment` type) or at least a `NonZeroU8`. Thoughts on this?

Reply:
Hmm... In `src/lib.rs`, at least, `Bytes` is used exclusively for 2, 4, and 8. Maybe an enum like `VectorSize` would suffice, with implementations of `Mul` with `Alignment`. Stuff like this would need to be changed to use a more relaxed type:

`src/back/glsl/mod.rs`
```
    /// A scalar with an unsupported width was requested.
    #[error("A scalar with an unsupported width was requested: {0:?} {1:?}")]
    UnsupportedScalar(crate::ScalarKind, crate::Bytes),
```

But that seems like an independent change. I think that would be best as a separate PR.